### PR TITLE
Alchemix Adapter added

### PR DESCRIPTION
**Pull request by 0xMars**

### CODE REVIEW

File: contracts/adapters/alchemix/AlchemixAdapter.sol [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+pragma solidity ^0.7.6;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../../interfaces/IUniswapV2Pair.sol";
+import "../../interfaces/IUniswapV2Router02.sol";
+import "../../libraries/UniswapV2Library.sol";
+import "../../BaseAdapter.sol";
+import "./IStakingPools.sol";
+
+contract LuaAdapter is BaseAdapter {
```
Comment (mdzor) [inline contracts/adapters/alchemix/AlchemixAdapter.sol:12]: Need to rename

**Comment by mdzor** [comment]
nice one! Could you add the pools in deploy/data/pools.json as well

**Review by mdzor — APPROVED** [review]

File: contracts/adapters/alchemix/AlchemixAdapter.sol [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    address constant MASTER_VAMPIRE =
+        0x12B7b9e21Ad9D7E8992e0c129ED0bccEaC185c3E;
+    IStakingPools constant ALCX_STAKINGPOOLS =
+        IStakingPools(0xAB8e74017a8Cc7c15FFcCd726603790d26d7DeCa);
+    IERC20 constant ALCX = IERC20(0xdBdb4d16EdA451D0503b854CF79D55697F90c8DF);
+    IUniswapV2Pair constant ALCX_WETH_PAIR =
+        IUniswapV2Pair(0x352E5EeE2D9C957710be656534D51Fbb3Ce074d6);
+
+    constructor(address _weth, address _factory) BaseAdapter(_weth, _factory) {}
```
Comment (aaronovz1) [inline contracts/adapters/alchemix/AlchemixAdapter.sol]: Update with latest changes in other adapters where master vampire address is passed as constructor arg

File: deploy/7.10.AlchemixAdapter.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  let { deployer, WETH, uniFactory } = await getNamedAccounts();
+
+  if (chainId == '1') {
+    const MasterVampire = await deployments.get('MasterVampire');
+
+    const AlchemixAdapter = await deploy('AlchemixAdapter', {
+      from: deployer,
+      log: true,
+      contract: 'AlchemixAdapter',
+      args: [WETH, uniFactory]
```
Comment (aaronovz1) [inline deploy/7.10.AlchemixAdapter.ts:22]: master vampire address needs to be passed as third arg

**Pull request merged**